src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolverEventSource.cs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) .NET Foundation. All rights reserved. Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using NuGet.Common;
+
+namespace Microsoft.Build.NuGetSdkResolver
+{
+    /// <summary>
+    /// Represents an <see cref="EventSource" /> for the NuGet-based MSBuild project SDK resolver.
+    /// </summary>
+    [EventSource(Name = "Microsoft-Build-NuGetSdkResolver")]
+    internal sealed class NuGetSdkResolverEventSource : EventSource
```

Comment:
I think it was someone from the VS Perf & Reliability team who once told me that Windows has a max number of ETW providers that can run at a time, so they very much discourage having fine-grained providers, as they've had to investigate customer issues where it caused problems.

Therefore, I propose that instead we create an ETW provider in NuGet.Common, that we can re-use throughout all of `src/NuGet.Core/*` projects (well, except for NuGet.Framework and NuGet.Versioning, but I don't see those assemblies emitting ETW events as useful).

Reply:
> I think it was someone from the VS Perf & Reliability team who once told me that Windows has a max number of ETW providers that can run at a time, so they very much discourage having fine-grained providers, as they've had to investigate customer issues where it caused problems.

Do you happen to know where we can verify this?  As far as I know, you can specify the exact providers you want which would essentially limit them per session.  But I'm interested to know where you heard this so we can decide what to do.  My preference is to have this new provider but I'm open to suggestion.

Reply:
I have moved this to `NuGet.Common`